client/iotcore/client.go

```diff
@@ -186,40 +186,40 @@ func (c *client) UpsertDevice(ctx context.Context,
 				_, err = svc.UpdateCertificate(ctx, paramsUpdateCertificate)
 			}
 		}
-
 		return awsDevice, err
-	} else if err == ErrDeviceNotFound {
-		err = nil
-	}
-
-	var resp *iot.CreateThingOutput
-	if err == nil {
-		resp, err = svc.CreateThing(ctx,
-			&iot.CreateThingInput{
-				ThingName: aws.String(deviceID),
-			})
+	} else if !errors.Is(err, ErrDeviceNotFound) {
+		return nil, fmt.Errorf("unexpected error getting the device: %w", err)
```

Comment:
why it was previously the other way around? it was ignoring the device not found error.